google/cloud/storage/internal/curl_request.h

```diff
@@ -25,73 +25,50 @@ namespace cloud {
 namespace storage {
 inline namespace STORAGE_CLIENT_NS {
 namespace internal {
-/**
- * Makes RPC-like requests using CURL.
- *
- * The Google Cloud Storage Client library using libcurl to make http requests,
- * this class manages the resources and workflow to make a simple RPC-like
- * request.
- */
+extern "C" std::size_t CurlRequestOnWriteData(char* ptr, size_t size,
+                                              size_t nmemb, void* userdata);
+extern "C" std::size_t CurlRequestOnHeaderData(char* contents, std::size_t size,
+                                               std::size_t nitems,
+                                               void* userdata);
+
 class CurlRequest {
  public:
-  CurlRequest();
-
+  CurlRequest() = default;
   ~CurlRequest() {
-    if (!factory_) {
-      return;
-    }
-    factory_->CleanupHandle(std::move(handle_.handle_));
-  }
-
-  CurlRequest(CurlRequest&& rhs) noexcept(false)
-      : url_(std::move(rhs.url_)),
-        headers_(std::move(rhs.headers_)),
-        user_agent_(std::move(rhs.user_agent_)),
-        response_payload_(std::move(rhs.response_payload_)),
-        received_headers_(std::move(rhs.received_headers_)),
-        logging_enabled_(rhs.logging_enabled_),
-        socket_options_(rhs.socket_options_),
-        handle_(std::move(rhs.handle_)),
-        factory_(std::move(rhs.factory_)) {
-    ResetOptions();
+    if (!factory_) return;
+    factory_->CleanupHandle(std::move(handle_));
```

Comment:
Nit: Perhaps `if (factory_) factory_->CleanupHandle(std::move(handle_));` might be clearer?

Reply:
Fixed.